### DISCUSSION TITLE
focal binaries are not working on ubuntu 20.04 LTS

### DISCUSSION THREAD

**Discussion by Thoulsadoom** [valkey-io/valkey#1076] [Q&A] [answered]
Hi guys,
I'm trying to use the provided binaries (https://download.valkey.io/releases/valkey-8.0.0-focal-x86_64.tar.gz) in a docker container with ubuntu 20.04 LTS. I stored the binaries in "/usr/local/bin".

However, when I try to start valkey-server im getting following error:
"error while loading shared libraries: libssl.so.3: cannot open shared object file: No such file or directory"

When I build my container with the sources (building sources in container and install them), valkey-server is able to start.
Am I missing something here in how to use the provided binaries?

**Comment by madolson** [comment]
@roshkhatri Can you take a look?

**Comment by roshkhatri** [comment, marked as answer]
The binaries provided on the website build with TLS support enabled (`make BUILD_TLS=yes`), you would have to add OpenSSL development libraries (e.g. libssl-dev on Debian/Ubuntu) to the docker in order to start valkey server.

**Reply by Thoulsadoom** [reply]
I guess I incorrectly assumed the focal binaries to be compatible with focal mirror dependencies. I now rely on building from source.

Thanks for the quick reply. 👍

**Reply by roshkhatri** [reply]
No worries, on that note we might re-think about  to make `the focal binaries to be compatible with focal mirror dependencies`.

**Reply by melroy89** [reply]
ps. The binaries provided are now also build with systemd support enabled.
